vispy/visuals/image.py

```diff
@@ -348,11 +348,17 @@ def clim(self):
     def clim(self, clim):
         if self._texture.set_clim(clim):
             self._need_texture_upload = True
-        # shortcut so we don't have to rebuild the whole color transform
-        if not self._need_colortransform_update:
-            self.shared_program.frag['color_transform'][1]['clim'] = self._texture.clim_normalized
+        self._update_colortransform_clim()
         self.update()
 
+    def _update_colortransform_clim(self):
+        # shortcut so we don't have to rebuild the whole color transform
+        try:
+            if not self._need_colortransform_update:
+                self.shared_program.frag['color_transform'][1]['clim'] = self._texture.clim_normalized
+        except RuntimeError:
+            return
```

Comment:
Maybe we can be a bit more specific and only put the `self._texture.clim_normalized` part in the try-except. Right now a RunTimeError raised by the program is also caught.